distributed/protocol/serialize.py

```diff
@@ -928,3 +928,8 @@ def register_generic(
     object_with_dict_serializer = ObjectDictSerializer(serializer_name)
     serialize_func.register(cls)(object_with_dict_serializer.serialize)
     deserialize_func.register(cls)(object_with_dict_serializer.deserialize)
+
+
+register_serialization_family("plasma", shared.ser_plasma, shared.deser_plasma)
+register_serialization_family("lmdb", shared.ser_lmdb, shared.deser_lmdb)
+register_serialization_family("vineyard", shared.ser_vineyard, shared.deser_vineyard)
```

Comment:
Am curious if just [setting `data` on the `Worker`]( https://github.com/dask/distributed/blob/bfc5cfea80450954dba5b87a5858cb2e3bac1833/distributed/worker.py#L484-L491 ), which in turn could [bypass usual spilling mechanisms]( https://github.com/dask/distributed/blob/bfc5cfea80450954dba5b87a5858cb2e3bac1833/distributed/worker_memory.py#L109-L114 ), and [disabling `compression`]( https://github.com/dask/distributed/blob/dd81b424971e81616e1a52fa09ce4698a5002d41/distributed/distributed.yaml#L208 ) would alleviate the need to hook these in this way while still getting much of the same benefit

Reply:
Gotcha couldn't this be useful though for redundancy? IOW if a worker dies, we could still recover the data stored in shared memory

That said, this is a good point about paying attention to what data is sent. Wonder if we should be thinking about shared memory as a comm

Admittedly hooking into serialization like this is pretty easy to do. Just thinking about how we might long term integrate some of these lessons to make shared memory easier to use for those that choose to use it

Reply:
>  IOW if a worker dies, we could still recover the data stored in shared memory

Either the scheduler would need to know that data was pushed to shm, or we'd need clever hashes of the buffers of a result; also, we need to make sure that the buffers are not purged because the worker that made them disappeared, before a new worker picks them up.

> thinking about shared memory as a comm

The thought had occurred, and that's sort of what vireyard's metadata service is. In fact, locally we could probably get better comms on linux sockets rather than TCP. But it was the simplicity of hijacking pickle at the point it is called that made me head in this direction.

Reply:
> Note that spilling might be available in the shm implementation, just haven't tried that yet.

FYI: Both plasma and vineyard supports implicit spilling (triggered when not enough memory for new allocation).

Reply:
Yes, I am aware of this, and we should endeavour to test it. I'm sure vineyard's is well documented, but https://arrow.apache.org/docs/python/plasma.html does not tell you how to activate plasma's.